src/ui/components/Library/Team/View/Tests/TestContextRoot.tsx

```diff
@@ -76,6 +77,7 @@ export function TestsContextRoot({ children }: { children: ReactNode }) {
         setFilterByText,
         testId: deferredTestId,
         testIdForDisplay: testId,
+        selectedTest: testId ? tests.find(t => t.testId === testId) ?? null : null,
```

Comment:
Pretty sure I said this on the other PR but maybe it didn't make it through for some reason.

Wrap this context value in memo (`value = useMemo(({...}), [...])`); this is just a good practice to avoid scheduling unnecessary udpates.